src/TroykaOLED.h

```diff
@@ -179,34 +76,55 @@ class TroykaOLED {
     uint8_t _i2cAddress;
     uint8_t _width;
     uint8_t _height;
-    uint8_t _numX;
-    uint8_t _numY;
+
+    struct {
+        int16_t x;
+        int16_t y;
+    } _last;
+
     uint8_t _codingName;
     bool _stateInvert;
     bool _stateAutoUpdate;
-    bool _stateImageBG;
-    uint8_t *_bufferDisplay;
+    uint8_t _imageColor;
+
+    union {
+        uint64_t column[HORIZONTAL_SIZE];
+        uint8_t asBytes[(VERTICAL_SIZE >> 3) * HORIZONTAL_SIZE];
+    } _screenBuffer;
+
     struct {
-        const uint8_t* fontData;
+        const uint8_t* data;
+        const uint8_t* remap;
         uint8_t width = 0;
         uint8_t height = 0;
-        uint8_t firstSymbol = 0;
-        uint8_t sumSymbol = 0;
-        uint8_t startSpace[3];
-        uint8_t sumSpace[3];
         bool invert = false;
-        bool background = true;
+        uint8_t color = WHITE;
         bool setFont = false;
     } _font;
 
-    void _drawPixel(int x, int y, uint8_t color = WHITE);
-    void _drawLine(int x1, int y1, int x2, int y2, uint8_t color = WHITE);
-    void _print(char*, int x, int y);
+    struct {
+        int16_t left;
+        int16_t right;
+    } changed;
+
+    void _drawPixel(int16_t x, int16_t y, uint8_t color = WHITE);
+    void _drawLine(int16_t x1, int16_t y1, int16_t x2, int16_t y2, uint8_t color = WHITE);
+    void _interpretParameters(int16_t x, int16_t y, int16_t w, int16_t h);
+    uint8_t _punchDigits(uint32_t n, char* d, uint8_t p, uint8_t b);
+    void _radixConverter(uint32_t num, bool sign, uint8_t bits, uint8_t base, int16_t x, int16_t y);
+    uint8_t _fontRemapping(char c);
+    void _print(char c, int16_t x, int16_t y);
+    void _print(char* s, int16_t x, int16_t y);
     char _itoa(uint8_t num);
-    char* _codingCP866(char* data);
+    char* _codingCP866(uint8_t* data);
```

Comment:
```suggestion
    char* _encodeCP866(uint8_t* data);
```

Methods should have a verb.

Reply:
Done: encodeToCP866